baker/buildLocalBake.ts

```diff
@@ -1,30 +1,58 @@
 #! /usr/bin/env node
 
 import parseArgs from "minimist"
-import { SiteBaker } from "./SiteBaker.js"
+import {
+    BakeStep,
+    BakeStepConfig,
+    bakeSteps,
+    SiteBaker,
+    validateBakeSteps,
+} from "./SiteBaker.js"
 import * as fs from "fs-extra"
 import { normalize } from "path"
+import { difference } from "lodash"
 
 const bakeDomainToFolder = async (
     baseUrl = "http://localhost:3000/",
-    dir = "localBake"
+    dir = "localBake",
+    bakeSteps?: Partial<BakeStepConfig>
 ) => {
     dir = normalize(dir)
     fs.mkdirp(dir)
-    const baker = new SiteBaker(dir, baseUrl)
+    const baker = new SiteBaker(dir, baseUrl, bakeSteps)
     console.log(
         `Baking site sans Wordpress with baseUrl '${baseUrl}' to dir '${dir}'`
     )
     await baker.bakeNonWordpressPages()
 }
 
 const args = parseArgs(process.argv.slice(2))
-const theArgs = args._
-// Usage: yarn buildLocalBake http://localhost:3000/ localBake
+const positionalArgs = args._
```

Comment:
It might be the right time to replace minimist with something more powerful. We'd get things like validation for free and it'd be finally clear what the command does. Here's a snippet with yargs

```typescript
#! /usr/bin/env node

import yargs from "yargs"
import { hideBin } from "yargs/helpers"
import { SiteBaker } from "./SiteBaker.js"
import * as fs from "fs-extra"
import { normalize } from "path"

const bakeDomainToFolder = async (
    baseUrl = "http://localhost:3000/",
    dir = "localBake"
) => {
    dir = normalize(dir)
    fs.mkdirp(dir)
    const baker = new SiteBaker(dir, baseUrl)
    console.log(
        `Baking site sans Wordpress with baseUrl '${baseUrl}' to dir '${dir}'`
    )
    await baker.bakeNonWordpressPages()
}

yargs(hideBin(process.argv))
    .command(
        "buildLocalBake [baseUrl] [dir]",
        "Bake the site to a local folder",
        (yargs) => {
            yargs
                .positional("baseUrl", {
                    type: "string",
                    default: "http://localhost:3000/",
                    describe: "Base URL of the site",
                })
                .positional("dir", {
                    type: "string",
                    default: "localBake",
                    describe: "Directory to save the baked site",
                })
                .option("steps", {
                    type: "array",
                    choices: ["assets", "charts", "variables"],
                    description: "Steps to perform during the baking process",
                })
        },
        async ({ baseUrl, dir, steps }) => {
            await bakeDomainToFolder(baseUrl, dir, steps)
            process.exit(0)
        }
    )
    .help()
    .alias("help", "h")
    .strict().argv
```

Other than that I really like this PR's simplicity / usefulness ratio!

Reply:
Nice, thanks @Marigold !

I had to duplicate the yargs code a couple more times because each one had a slightly different usecase, but overall I think we're still getting quite a lot of bang for our complexity buck.

I've tested `buildLocalBake` and `buildAndDeployToSite` with various combinations of steps and flags. 

Feel free to have a test yourself and let me know how you find it!